src/main.rs

```diff
@@ -601,6 +601,9 @@ pub async fn verify_from_repo(
     temp_dir_opt: &mut Option<String>,
 ) -> anyhow::Result<()> {
     if remote {
+        if connection_url != Some("m".to_string()) {
+            return Err(anyhow!("Remote verification only works with mainnet. Include -um flag to verify on mainnet."));
+        }
```

Comment:
You should hit the `get_genesis_hash` rpc endpoint and check that it matches the appropriate hash. They don't necessarily need to use `-um`

Reply:
Updated with these changes.